web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts

```diff
@@ -236,6 +243,63 @@ export function useCreateDatabase() {
     setPollActive(true);
   }
 
+  async function enableAutoDiscovery({
+    selectedRegion,
+  }: {
+    selectedRegion: Regions;
+  }) {
+    setAttempt({ status: 'processing' });
+
+    // Create a discovery config for the discovery service.
+    let discoveryConfig = autoDiscoveryCfg;
+    try {
+      if (!discoveryConfig) {
+        discoveryConfig = await createDiscoveryConfig(clusterId, {
+          name: crypto.randomUUID(),
+          discoveryGroup: DISCOVERY_GROUP_CLOUD,
+          aws: [
+            {
+              types: ['rds'],
+              regions: [selectedRegion],
+              tags: { '*': ['*'] },
+              integration: (agentMeta as DbMeta).integration.name,
+            },
+          ],
+        });
+        setAutoDiscoveryCfg(discoveryConfig);
+      }
+    } catch (err) {
+      handleRequestError(err, 'failed to create discovery config: ');
+      return;
+    }
+
+    // See if we have existing database services that has asteriks or region
+    try {
+      const { services } = await ctx.databaseService.fetchDatabaseServices(
+        clusterId
+      );
```

Comment:
hrm, should i try using `Promise.allSettled` instead? (though this may potentially take longer if there was an error with one)

it's because with `Promise.all` i don't think i can tell tell which promise failed and sometimes the backend returns confusing messages and we emit all errors:

```
function handleRequestError(err: Error, preErrMsg = '') {
    const message = getErrMessage(err);
    setAttempt({ status: 'failed', statusText: `${preErrMsg}${message}` });
    emitErrorEvent(`${preErrMsg}${message}`);
  }
```